src/hardware/serialport/softmodem.cpp

```diff
@@ -288,6 +288,13 @@ void CSerialModem::SendRes(const ResTypes response) {
 }
 
 bool CSerialModem::Dial(const char * host) {
+	// Close the server socket before dialing
+	waitingclientsocket.reset(nullptr);
+	if (serversocket) {
+		serversocket->Close();
+		serversocket.reset(nullptr);
+	}
```

Comment:
Looking good :+1: 

One last suggestion (just style) from me is to use `.reset()` instead of with `nullptr`, because that's how all the other sources do it (I did a `git grep '.reset('` to show me all the instances)

Reply:
Good point, after that it's good to go.

Reply:
Every `reset` instance in that file does `reset(nullptr)`. Unless you want me to do refactoring in this file, I think it's best to leave it as is.

Reply:
Yup - I would say mass replace 'em if we're changing that code anyway (can be a separate commit` "Make socket resets consistent"` or something like that). 

@johnnovak's always suggested these type of consistency improvements in my PRs, might as well do it. It's a thumbs up from me :smile:

Reply:
@NicknineTheEagle Yeah, can you please mass-replace `reset(nullptr)` with `reset()` in a separate commit, thanks. They only occur in this file, nowhere else:

<img width="732" alt="image" src="https://github.com/user-attachments/assets/ac0b2185-8dff-4bb0-9996-32b19525373d">